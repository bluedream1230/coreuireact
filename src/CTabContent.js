import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { mapToCssModules } from './Shared/helper.js'
//component - CoreUI / CTabContent

const CTabContent = props => {

  const {
    className,
    cssModule,
    //
    innerRef,
    ...attributes
  } = props;

  // render
  const classes = mapToCssModules(classNames('tab-content',className),cssModule)

  return (
    <div className={classes} {...attributes} ref={innerRef}/>
  )
}

CTabContent.propTypes = {
  className: PropTypes.string,
  cssModule: PropTypes.object,
  //
  innerRef: PropTypes.oneOfType([PropTypes.object, PropTypes.func, PropTypes.string]),
};

export default CTabContent
