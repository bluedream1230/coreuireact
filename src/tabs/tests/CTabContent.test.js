import React from 'react'
import { configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer';

import CTabContent from '../CTabContent'

configure({ adapter: new Adapter() })

describe('CTabContent', () => {
  it('renders basic wrapper correctly', () => {
    const component = renderer.create(<CTabContent/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot()
  })
  it('renders customized wrapper correctly', () => {
    const componentCustomized = renderer.create(
      <CTabContent
        className='class-name'
      >
        CTabContent
      </CTabContent>
    );
    let tree = componentCustomized.toJSON();
    expect(tree).toMatchSnapshot()
  })
})