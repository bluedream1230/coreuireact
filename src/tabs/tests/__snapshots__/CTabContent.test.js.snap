// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTabContent renders basic wrapper correctly 1`] = `
<div
  className="tab-content"
/>
`;

exports[`CTabContent renders customized wrapper correctly 1`] = `
<div
  className="tab-content class-name"
>
  CTabContent
</div>
`;
